engine/engine_test.go

```diff
@@ -2980,6 +2980,13 @@ func TestInstantQuery(t *testing.T) {
 			queryTime: time.Unix(0, 0),
 			query:     "sort_desc(http_requests_total)",
 		},
+		{
+			name: "histogram_quantile with mock duplicate labels",
```

Comment:
What does mock duplicate labels means in this context?

Reply:
"-X or X" will have name labels dropped from the left and not from the right but its kinda mock because it never can return both. It just illustrates the limitation of calculating all possible series beforehand; thats why we need to calculate duplicate labels in the engine and not in the operators.